digester/main.go

```diff
@@ -19,6 +66,65 @@ func CheckError(err error) {
 	}
 }
 
+func unzipFile(file *zip.File, dest string) error {
+	// Check if file paths are not vulnerable to Zip Slip
+	filePath := filepath.Join(dest, file.Name)
+	if !strings.HasPrefix(filePath, filepath.Clean(dest)+string(os.PathSeparator)) {
+		return fmt.Errorf("%s: illegal file path", filePath)
+	}
+
+	if file.FileInfo().IsDir() {
+		if err := os.MkdirAll(filePath, os.ModePerm); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if err := os.MkdirAll(filepath.Dir(filePath), os.ModePerm); err != nil {
+		return err
+	}
+
+	destFile, err := os.OpenFile(filePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, file.Mode())
+	if err != nil {
+		return err
+	}
+	defer destFile.Close()
+
+	zipFile, err := file.Open()
+	if err != nil {
+		return err
+	}
+	defer zipFile.Close()
+
+	if _, err := io.Copy(destFile, zipFile); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func unzipSource(src, dest string) error {
+	reader, err := zip.OpenReader(src)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+
+	dest, err = filepath.Abs(dest)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range reader.File {
+		err := unzipFile(file, dest)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 func main() {
 	err := godotenv.Load(".env")
```

Comment:
This should not be required (and won't work either) as the env vars are set by docker-compose.

Reply:
fallback daala tha but who wants it